xtask/codegen/src/js_kinds_src.rs

```diff
@@ -508,6 +508,8 @@ pub const JS_KINDS_SRC: KindsSrc = KindsSrc {
         "JSX_EXPRESSION_CHILD",
         "JSX_SPREAD_CHILD",
         "JSX_STRING",
+        // Grit metavariable
+        "JS_GRIT_METAVARIABLE",
```

Comment:
Why was this added here?

Reply:
I have some thoughts I don't know if they fit here. If we are to support [embedded language formatting](https://github.com/biomejs/biome/discussions/2227) in the future, we may also need a "metavaraiable" kind node. For example, in JavaScript (there may be similar concepts in other languages), the embedded language are usually embedded as a template string literal that can optionally hold expression placeholders. When we want to parse and format the template literal,  I think we can see those placeholders as metavariables in our internal structure.

```ts
const foo = JSON.stringify("bar");
const snippet = js`console.log(${ foo })`; // the "${ foo }" is a placeholder
```

So I think this kind of node is necessary but I just don't know if we should have `GRIT` in the name.

Reply:
That’s a very insightful comment, @Sec-ant , thanks! There is definitely a parallel between metavariables as used in Grit snippets and embedded expressions inside template literals, even though the syntax is different.

I would be happy to revise this (in a separate PR) if we can reuse these node kinds for that. Definitely something to keep in mind!